app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/HeartRateUtils.java

```diff
@@ -28,7 +32,42 @@
      */
     public static final int MAX_HR_MEASUREMENTS_GAP_MINUTES = 10;
 
-    public static boolean isValidHeartRateValue(int value) {
-        return value > HeartRateUtils.MIN_HEART_RATE_VALUE && value < HeartRateUtils.MAX_HEART_RATE_VALUE;
+    private int maxHeartRateValue;
+    private int minHeartRateValue;
+
+    /**
+     * extra class used for synchronized access of the singleton
+     */
+    private static final class HeartRateUtilsInstanceHolder {
+        static final HeartRateUtils INSTANCE = new HeartRateUtils();
+    }
+
+    public static HeartRateUtils getInstance() {
+        return HeartRateUtilsInstanceHolder.INSTANCE;
+    }
+
+    /**
+     * Singleton - to access this class use the static #getInstance()
+     */
+    private HeartRateUtils() {
+        updateCachedHeartRatePreferences();
+    }
+
+    public void updateCachedHeartRatePreferences(){
+        Prefs prefs = GBApplication.getPrefs();
+        maxHeartRateValue = prefs.getInt("chart_max_heart_rate", MAX_HEART_RATE_VALUE);
+        minHeartRateValue = prefs.getInt("chart_min_heart_rate", MIN_HEART_RATE_VALUE);
```

Comment:
Can you add those preference keys to `GBPrefs` and use them instead of scattering the keys? I know we have more of those, but I'd like to have fewer of them, not more :-)

Reply:
Ok did that.